### ISSUE TITLE
TO Golang API: Routing bugs

### ISSUE THREAD

**Issue by moltzaum** [apache/trafficcontrol#2889]
There seem to be two bugs:
1) no reverse proxy back to perl
2) instead of a 404, getting a 401 instead

I am on the master branch.

`curl -XPOST --cookie $mc -Lvsk 'https://localhost:6443/api/1.1/users' -d @user.json | jq`
```
{
  "alerts": [
    {
      "level": "error",
      "text": "Unauthorized, please log in."
    }
  ]
}
```

Curling an invalid endpoint such as /api/1.3/blah also returns a 401 when it should return a 404.

**Comment by moltzaum** [comment]
I'm setting the cookie the same way I always have, and can hit the 1.4 golang endpoint that I was writing just fine. I think @rob05c ran into the same issue. From your comment, it seems everything is working as expected for you?

**Comment by rawlinp** [comment]
Yeah I'm currently on d8cf653eaecb0074db8a860ba9393982d8ff5cad and getting 404s when hitting nonexistent routes when logged in, 401s when _not_ logged in, and was also able to create a user using the API while logged in.

**Comment by mitchell852** [comment]
6443 is the port where TO golang is running?

**Comment by moltzaum** [comment]
Update:
I have completely purged and reset my docker environment, rebuilt the TO rpm from master, and I am still running into this issue. Any golang endpoint I can hit properly, and every perl endpoint returns a 401 (with the same cookie). Putting in endpoints that should be a 404 return 401 if at the very least the version number in the url is given (eg. /api/blah does return a 404, but /api/1.1 returns a 401)

6443 is the golang port, and should reverse proxy to perl

**Comment by mitchell852** [comment]
@moltzaum - did u figure out the problem?

**Comment by jrushford** [comment]
I'm seeing the same problem with grovetccfg which uses the client,

I get a 401 unauthorized, 

"https://trafficops.comcast.net/api/1.4/user/login {"alerts":[{"level":"error","text":"Unauthorized, please log in."}]}"
